mlir/lib/Dialect/Affine/Analysis/CMakeLists.txt

```diff
@@ -8,6 +8,9 @@ add_mlir_dialect_library(MLIRAffineAnalysis
   ADDITIONAL_HEADER_DIRS
   ${MLIR_MAIN_INCLUDE_DIR}/mlir/Dialect/Affine
 
+  DEPENDS
+  MLIRFuncOpsIncGen
```

Comment:
This does not seem like the right dependency to me: we should depend on the FuncDialect here. The IncGen targets are meant to be depending on by the dialects themselves.

You can also see this as a transitive dependency issue: you're skipping one step in the dependency chain.